sdk/eventhub/event-hubs/src/eventPosition.ts

```diff
@@ -226,3 +83,76 @@ export function isLatestPosition(eventPosition: EventPosition): boolean {
 
   return false;
 }
+
+/**
+ * Gets the `EventPosition` corresponding to the location of the the first event present in the partition.
+ * Pass this position to the `EventHubConsumerClient.subscribe()` method to begin receiving events from the
+ * first event in the partition which has not expired due to the retention policy.
+ */
+export const earliestEventPosition: EventPosition = {
+  offset: -1
+};
+
+/**
+ * Gets the `EventPosition` corresponding to the end of the partition.
+ * Pass this position to the `EventHubConsumerClient.subscribe()` method to begin receiving events from the
+ * event that is enqueued right after the method call.
+ * @returns EventPosition
+ */
+export const latestEventPosition: EventPosition = {
+  offset: "@latest"
+};
+
+/**
+ * @ignore
+ * @internal
+ */
+export function validateEventPositions(
+  position: EventPosition | { [partitionId: string]: EventPosition }
+) {
+  if (position == undefined) {
+    return;
+  }
+
+  const offsetPresent = position.offset != undefined;
+  const sequenceNumberPresent = position.sequenceNumber != undefined;
+  const enqueuedOnPresent = position.enqueuedOn != undefined;
+  const keys = Object.keys(position);
+
+  if (offsetPresent || sequenceNumberPresent || enqueuedOnPresent || !keys.length) {
+    validateEventPosition(position);
+    return;
+  }
+
+  const positions = position as { [partitionId: string]: EventPosition };
+  for (let i = 0; i < keys.length; i++) {
+    if (positions.hasOwnProperty(keys[i])) {
```

Comment:
```suggestion
    if (Object.prototype.hasOwnProperty.call(positions, keys[i])) {
```
Since `positions` is being provided by the user and not created by the SDK, the above change may be safer. It's possible to create an object that lacks a real prototype by calling `Object.create(null)`, which would cause the current expression to throw an error.